### PR TITLE
runplus: add option to always resume

### PR DESCRIPTION
**Pull request by bobrippling**
For example, when doing reps with 1 minute rests, a user can configure runplus so it automatically resumes if started within the minute

### CODE REVIEW

**Comment by bobrippling** [comment]
@halemmerich @thyttan does this seem ok to you both? I think you're the main users of `runplus`

**Comment by halemmerich** [comment]
I'm not quite sure what you mean by this. Is it auto resume after manually stopping for a slow/rest period? The target would be to have only the high intensity periods tracked/shown?

**Comment by bobrippling** [comment]
Yeah exactly, like if you're doing reps with a lot of standing about in between - I found I'd have to press start then wait and tap the resume button, sometimes forgetting during a session

**Comment by thyttan** [comment]
By automatically, it's not that it sets a timer and starts by itself after i runs out? Just that the menu is not displayed within the window of the timeout and the records will be appended to the last log?

**Comment by bobrippling** [comment]
Yeah that's poor choice of phrasing on my part - it's exactly the existing behaviour, just with the ability to customise the time period (so like you say, currently the user is always prompted to resume the run or start anew after 10s)

**Comment by halemmerich** [comment]
How about letting a short button press pause and resume (including showing paused as state in the bottom right) and longer button press stop the recording? That way there is no additional setting and it is always possible to actually end the run. It would allow more flexible intervals.
Maybe even write a line into the recorder log when the pause starts and ends instead of actually pausing the recorder to make later analysis easier. That would allow to check for pulse recovery etc. during the rest intervals.
Thinking about it, maybe short press could just signify a new interval/round start/stop and otherwise everything would just run like normal? The UI could do whatever it wants with the intervals, i.e. reset exstats, show averages/values only for the last interval or whatever.

**Comment by halemmerich** [comment]
How does actually stopping work? Say I configure a 5 minute auto resume time, do I have to kill `runplus` and go to the recorder to stop the run?

**Comment by bobrippling** [comment]
For a 5 minute resume time, yes, you'd have to restart `runplus` - but that'd be it, it would then kick off a new recording (in fact you'd get a prompt - see below snippets), you wouldn't need to tangle with the recorder app manually:

https://github.com/espruino/BangleApps/blob/e856e381fad97cea0c88acd5cbb78c448c1abe0b/apps/runplus/app.js#L84-L91

https://github.com/espruino/BangleApps/blob/e856e381fad97cea0c88acd5cbb78c448c1abe0b/apps/recorder/widget.js#L267-L273

**Comment by halemmerich** [comment]
So if a later rework to include laps or other handling of the start/stop/pause/resume mechanics is still on the table I'm fine with this change. Maybe we should signify if the timer has run out in the UI by replacing the STOP by PAUSE or something like that as long as it would autoresume on button press.

**Comment by bobrippling** [comment]
Sounds good, sorted the timer UI (dbc9833), let me know what you think

File: apps/runplus/app.js [hunk 1]
```diff
@@ -49,9 +54,15 @@ let statIDs = [settings.B1,settings.B2,settings.B3,settings.B4,settings.B5,setti
...
 // ---------------------------
 
+function inPauseWindow() {
+  return exs.state.duration <= settings.resume.promptAfter;
+}
+
 function setStatus(running) {
-  layout.button.label = running ? "STOP" : "START";
-  layout.status.label = running ? "RUN" : "STOP";
+  const paused = isInPauseWindow();
```
Comment (thyttan) [inline apps/runplus/app.js:62]: ... this doesn't match :)

File: apps/runplus/app.js [hunk 2]
```diff
@@ -49,9 +54,15 @@ let statIDs = [settings.B1,settings.B2,settings.B3,settings.B4,settings.B5,setti
 let exs = ExStats.getStats(statIDs, settings);
 // ---------------------------
 
+function inPauseWindow() {
```
Comment (thyttan) [inline apps/runplus/app.js:57]: This and ...

Comment (thyttan) [inline apps/runplus/app.js:57]: This and ...

File: apps/runplus/app.js (hunk 1 above)
Comment (thyttan) [inline apps/runplus/app.js:62]: ... this don't match :)

Comment (bobrippling) [reply apps/runplus/app.js]: That's what I get for last minute changes! Thank you

**Comment by bobrippling** [comment]
I've played around with this more and I think the timeout isn't worth the complexity, my use case would be to have the timeout be ridiculously high anyway - I've changed it to a simple toggle.

Given this a test and it looks to work well, let me know what you think :)

**Comment by thyttan** [comment]
Thanks!

**Comment by bobrippling** [comment]
Thanks for the review/merge!

**Pull request merged**